processor/spanintentprocessor/processor.go

```diff
@@ -0,0 +1,448 @@
+	}
+	mSampledCacheMisses, err := meter.Int64Counter("processor.spanintent.sampled_cache_misses")
+	if err != nil {
+		return nil, err
+	}
+	mUnsampledCacheHits, err := meter.Int64Counter("processor.spanintent.unsampled_cache_hits")
+	if err != nil {
+		return nil, err
+	}
+	mUnsampledCacheMisses, err := meter.Int64Counter("processor.spanintent.unsampled_cache_misses")
+	if err != nil {
+		return nil, err
+	}
+	mTracesClassifiedTotal, err := meter.Int64Counter("processor.spanintent.traces_classified_total")
+	if err != nil {
+		return nil, err
+	}
+	mTracesSampled, err := meter.Int64Counter("processor.spanintent.traces_sampled")
+	if err != nil {
+		return nil, err
+	}
+	mTracesUnsampled, err := meter.Int64Counter("processor.spanintent.traces_unsampled")
+	if err != nil {
+		return nil, err
+	}
+	mErrorsTotal, err := meter.Int64Counter("processor.spanintent.errors_total")
+	if err != nil {
+		return nil, err
+	}
+	mProcessingDuration, err := meter.Int64Histogram("processor.spanintent.processing_duration_ms")
+	if err != nil {
+		return nil, err
+	}
+	mSamplingDecisionLatency, err := meter.Int64Histogram("processor.spanintent.sampling_decision_latency_us")
+	if err != nil {
+		return nil, err
+	}
+
+	sampledCache, err := cache.NewCache[bool](cfg.SampledTracesCacheSize)
+	if err != nil {
+		return nil, err
+	}
+	unsampledCache, err := cache.NewCache[bool](cfg.UnsampledTracesCacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	return &spanIntentProcessor{
+		logger:          logger,
+		cfg:             cfg,
+		nextConsumer:    nextConsumer,
+		traceDataBuffer: make(map[pcommon.TraceID]*traceData),
+		tdigestMap:      make(map[string]*utility.TDigest),
+		quantileEMAMap:  make(map[string]*quantileEMA),
+		emaAlpha:        0.8, // smoothing factor for latency categorization
+		sampledTraces:   sampledCache,
+		unsampledTraces: unsampledCache,
+		stopCh:          make(chan struct{}),
+
+		mSpansReceived:           mSpansReceived,
+		mNewTraceIDReceived:      mNewTraceIDReceived,
+		mTraceBufferSize:         mTraceBufferSize,
+		mSampledCacheHits:        mSampledCacheHits,
+		mSampledCacheMisses:      mSampledCacheMisses,
+		mUnsampledCacheHits:      mUnsampledCacheHits,
+		mUnsampledCacheMisses:    mUnsampledCacheMisses,
+		mTracesClassifiedTotal:   mTracesClassifiedTotal,
+		mTracesSampled:           mTracesSampled,
+		mTracesUnsampled:         mTracesUnsampled,
+		mErrorsTotal:             mErrorsTotal,
+		mProcessingDuration:      mProcessingDuration,
+		mSamplingDecisionLatency: mSamplingDecisionLatency,
+	}, nil
+}
+
+type quantileEMA struct {
+	Q75         float64
+	Q95         float64
+	Initialized bool
+}
+
+func (p *spanIntentProcessor) Start(ctx context.Context, host component.Host) error {
+	// go p.runTickLoop()
+	p.logger.Info("Processor Started: spanintentprocessor")
+	return nil
+}
+
+func (p *spanIntentProcessor) Shutdown(ctx context.Context) error {
+	close(p.stopCh)
+	return nil
+}
+
+func (p *spanIntentProcessor) init() {
```

Comment:
When to call this init() function. I do not see any reference to it, or did I miss something?